terraform/provisioning/common.bash

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+common::exit_error() {
+  local -r message="${1:?}"
+  local -r code="${2:-1}"
+  echo "${message}" >&2
+  exit "${code}"
+}
+
+common::caddy_write_config() {
+  local -r domain="${1}"
+  sudo mkdir -p /etc/caddy
+  sudo mkdir -p /var/www
+  cat <<EOF | sudo tee /etc/caddy/Caddyfile
+${domain} {
+  proxy  / http://localhost:10000 {
+    transparent
+    websocket
+  }
+  root   /var/www/
+  log    stdout
+  errors stdout
+}
+EOF
+}
+
+common::caddy_run() {
+  sudo mkdir -p /var/lib/caddy/dotcaddy
+  sudo docker run -d \
+    --name caddy \
+    --network host \
+    --volume=/etc/caddy/Caddyfile:/etc/Caddyfile:ro \
+    --volume=/var/lib/caddy/dotcaddy:/root/.caddy:rw \
+    quay.io/schu/caddy:0.11.5
+}
+
+common::contained.af_run() {
+  sudo docker run -d \
+    --name contained.af \
+    --network host \
+    quay.io/schu/contained.af \
```

Comment:
The container image should probably come from a terraform variable instead of hardcoding `schu`.

ditto line 38

It could be done in a later PR.

Reply:
Not sure, let's defer the decision to later.